### DISCUSSION TITLE
Does the multi select support being dependent?

### DISCUSSION THREAD

**Discussion by BrendanBasson4** [filamentphp/filament#1398] [Help] [answered]
I'm trying to populate a multi selects options via another selects value but can't seem to get it working.<br>
The selects options don't always populate on the first selection they populate on the second selection but then they can't be selected and won't change when a new selection is made.<br>

### Properties

```php
public $jobFunctionId;
public $subJobFunctions;
```

```php
$this->form->fill(
[
     'jobFunctionId' => $this->jobFunctionId,
]);
```

### Selects

```php
Card::make([
  Select::make('jobFunctionId')
    ->label('Job Functions')
    ->options(JobFunction::all()->pluck('name', 'id')->toArray())
    ->required()
    ->reactive()
    ->columnSpan(3),

  MultiSelect::make('subJobFunctions')
    ->label('Sub Job Functions')
    ->options(
      function(callable $get){
        $jobFunctionId = $get('jobFunctionId');
          if (!$jobFunctionId) {
            return [];
          } else {
            $jobFunction = JobFunction::find($jobFunctionId);
            return $jobFunction->sub_job_functions->pluck('name', 'id');
          }
        }
    )
    ->reactive()
    ->required()
    ->columnSpan(3),
])->columns(6)
```
### Example
![ezgif com-gif-maker](https://user-images.githubusercontent.com/47768746/151751541-470a284d-cf8d-4e67-a341-9a5d2a939b88.gif)

**Comment by danharrin** [comment]
They do support being dependant - but not with `->options()`. Instead, use `->getSearchResultsUsing()` and `->getOptionLabelsUsing()` (both methods are documented in the MultiSelect docs). This is because Filament is unaware if the `->options()` change between requests (this is userland data) so the options need to be dynamically populated via search on-demand instead.

**Reply by BrendanBasson4** [reply]
Thank you it works perfectly

**Comment by danharrin** [comment, marked as answer]
As per #1951, I've found a solution. You may now have dynamic, dependant `options()`, with full search support :)
